### PR TITLE
Fixing lift-over (again) WRT genotypes

### PR DESCRIPTION
**Pull request by yfarjoun**
### Description

LiftOverVcf didn't do a good job lifting over genotypes in simple (non-flipped indel, and all SNPs) cases. 

This was discovered thanks to @bbimber who opened a pr #929 which I closed since the stack trace shown was out-of-date. I hope that this PR revoles the issue, but if not, @bbimber is invited to re-open that PR, or open a new issue documenting the issue at hand.

@bbimber's original description, that I neglected to copy here:
...while picard tries to update the alleles in genotypes for reverse complementation, it does not update the alleles in VariantContextBuilder to reflect any of those changes. This PR applies similar logic to the set of Ref/Alt alleles as already happens for the genotype alleles.

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

**Comment by bbimber** [comment]
I believe this covers the concerns I had in my earlier PR.  This pre-dates this change; however, line 436:

reverseComplementAlleleMap.clear();

in flipIndel() doesnt seem to be necessary.

**Comment by bbimber** [comment]
@yfarjoun, please see #934

**Comment by yfarjoun** [comment]
@bbimber good now?

File: src/main/java/picard/vcf/LiftoverVcf.java [hunk 1]
```diff
@@ -387,12 +383,15 @@ protected static VariantContext liftSimpleVariant(final VariantContext source, f
             return null;
         }
         final List<Allele> alleles = new ArrayList<>();
+        final Map<Allele, Allele> reverseComplementAlleleMap = new HashMap<>(2);
+
```
Comment (eitanbanks) [inline src/main/java/picard/vcf/LiftoverVcf.java]: There's an extra space here and on line 409 below.

**Comment by yfarjoun** [comment]
OK?

**Comment by bbimber** [comment]
If that's directed at me, yes, seems good.

**Comment by eitanbanks** [comment]
Same with me

**Pull request merged**